### PR TITLE
Partest recognizes using

### PR DESCRIPTION
**Pull request by som-snytt**
Require line comment for current style pragma.

Other scala-cli style pragmas could be supported, but only option is recognized.

This simplifies posting code from reports, where dependencies are not involved.

### CODE REVIEW

File: src/partest/scala/tools/partest/nest/Runner.scala [hunk 1]
```diff
@@ -480,11 +480,19 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
...
-      val lines = header.filter(_.contains(tag)).map(line => stripped(line).trim())
-      if (name == ToolName.filter) lines else lines.flatMap(tokenize)
+      val namePattern = raw"\s*//\s*$name:\s*(.*)".r
+      val optionsPattern = raw"\s*//>\s*using\s+option(s)?\s+(.*)".r
+      def matchLine(line: String): List[String] =
+        line match {
+          case namePattern(rest) => if (name == ToolName.filter) List(rest) else tokenize(rest)
+          case _ if name == ToolName.scalac =>
+            line match {
+              case optionsPattern(plural, rest) => if (plural == null) List(rest) else rest.split(',').toList.map(_.trim)
```
Comment (lrytz) [inline src/partest/scala/tools/partest/nest/Runner.scala]: maybe for trailing whitespace?

```suggestion
              case optionsPattern(plural, rest) => if (plural == null) List(rest.trim) else rest.split(',').toList.map(_.trim)
```

Comment (som-snytt) [reply src/partest/scala/tools/partest/nest/Runner.scala]: Now that you mention it, one reason to prefer `// scalac: args` is that it uses `tokenize`. `//> using options x, y, z` is actually unfortunate. `//> using options -Wconf:cat=deprecation:s,cat=feature:e` what does it do?

I will trim for now and explore the complications later.

I was glad to use one standard for `tokenize` everywhere, but like the xkcd, let's standardize on scala-cli and now we have two standards.

Comment (som-snytt) [reply src/partest/scala/tools/partest/nest/Runner.scala]: I guess they support double-quotes. Maybe sufficient to tokenize then strip trailing comma, also simple comma.
```
tokenize("""-Xlint, "-Wconf:cat=deprecation:s,cat=feature:e", -feature , -deprecation""")
```

Comment (som-snytt) [reply src/partest/scala/tools/partest/nest/Runner.scala]: Not trimming was cut/paste from `filter`, which probably should trim and not take the raw line. The filter would inadvertently not match with spurious spaces.

**Comment by som-snytt** [comment]
Rebased and squashed, with improved handling of options. To avoid adding LOC, use tokenize but strip commas. Probably the limitation is that args can't end in comma.

The null test on optional group extraction is too bad. By coincidence, `regsafe` was just mentioned on dotty because it broke in the CB. Recall that `regextractor` would supply `Option` a decade ago. It's too bad that `regsafe` says that no one care about nulls, and also too bad that I'm checking for null in 2023.

**Review by lrytz — APPROVED** [review]

**Comment by som-snytt** [comment]
Or follow up with https://github.com/VirtusLab/using_directives thanks @SethTisue for the tip.

**Pull request merged**